tests/foundry/client/data/documents/macro.test-d.ts

```diff
@@ -5,7 +5,10 @@ const macro = new Macro({ name: "my macro", scope: "global", type: "script" });
 
 // properties and functions added by the concrete `Macro` class
 expectTypeOf(macro.isAuthor).toEqualTypeOf<boolean>();
-expectTypeOf(macro.execute).toEqualTypeOf<(scope?: { actor?: Actor; token?: Token }) => any>();
+expectTypeOf(await macro.execute({ effect: new ActiveEffect() })).toEqualTypeOf<unknown>();
+
+// @ts-expect-error The actor property must be an actual Actor
+expectTypeOf(await macro.execute({ actor: new ActiveEffect() })).toEqualTypeOf<unknown>();
```

Comment:
Can you add two more tests?:
1. Running this with additional arguments to add to the macro scope.
2. Running this with a macro of type `"chat"` so that it returns void.

Reply:
Good call on the second test, even with `if (chatMacro.type === "chat") expectTypeOf(chatMacro.execute()).toEqualTypeOf<void>();` it's still failing the test

Reply:
Refining the return type `("chat" extends this["type"] ? void : never) | ("script" extends this["type"] ? Promise<unknown> : never)` gets me *closer*; it starts as `void | Promise<unknown>` and is narrowed by checking the type for script, but checking the type for chat does not. I assume this is a particular property of the void return type?